.circleci/config.yml

```diff
@@ -38,29 +33,93 @@ jobs:
             docker-compose build frontend 
             docker-compose build py-backend
 
-  lint:
+  test:
     docker:
-      - image: cimg/python:3.9.7
+      - image: cimg/base:2021.05
     steps:
       - checkout
+      - setup_remote_docker:
+          version: 19.03.13
       - run:
-          name: Run linter checks
+          name: Create env file
           command: |
-            pip install black isort
-            echo "Running black checks..."
-            black backend/python/app/ --check
-            echo "Checking import order..."
-            isort --profile black backend/python/app/ -c
-            echo "Code properly linted!"
+            touch .env
+            echo "Writing env variables to .env"
+            env > .env
+      - run:
+          name: Running tests
+          command: |
+            set -x
+            echo "Running backend tests"
+            docker-compose -f docker-compose.testing.yml up --abort-on-container-exit
+
+  deploy:
+    docker:
+      - image: cimg/node:14.15.5
+    steps:
+      - checkout
+      - run:
+          name: Build frontend
+          command: |
+            cd frontend
+            yarn install --frozen-lockfile
+            yarn build
+      - run:
+          name: Setup Credentials
+          command: |
+            sudo apt-get update
+            sudo apt-get -qq install git-ftp
+            echo "Setting up FTP credentials..."
+            HOST=`echo ${FTP_HOST_BASE64} | base64 --decode`
+            USERNAME=`echo ${FTP_USERNAME_BASE64} | base64 --decode`
+            PASSWORD=`echo ${FTP_PASSWORD_BASE64} | base64 --decode`
+            echo "export HOST=$HOST" >> $BASH_ENV
+            echo "export USERNAME=$USERNAME" >> $BASH_ENV
+            echo "export PASSWORD=$PASSWORD" >> $BASH_ENV
+      - run:
+          name: Deploy frontend
+          command: |
+            echo "Deploying frontend..."
+            git config git-ftp.url ${HOST}
+            git config git-ftp.user ${USERNAME}
+            git config git-ftp.password ${PASSWORD}
+            git ftp init --insecure --syncroot frontend/build
+            echo "Frontend deployed!"
+      - run:
+          name: Deploy backend
+          command: |
+            echo "Deploying backend..."
+            git config git-ftp.url "${HOST}/~/backend/"
+            git config git-ftp.user ${USERNAME}
+            git config git-ftp.password ${PASSWORD}
+            git ftp init --insecure --syncroot backend/python
+            echo "Backend deployed!"
 
 workflows:
   version: 2
   lint-build-test:
     jobs:
-      - lint
+      - lint:
```

Comment:
why add the version tag filter to `lint` `build` and `test?`

Reply:
for some strange reason, because `lint`, `build` and `test` are all dependencies of `deploy`, the version tag filter doesn't work unless it's added to all of them
https://discuss.circleci.com/t/how-to-deploy-only-on-tagged-and-master-commits/25581/2

Reply:
ohhh makes sense. I was confused about how filters-tags work. For anyone who reads this in the future, this is relevant: 
https://circleci.com/docs/2.0/configuration-reference/#tags

> CircleCI does not run workflows for tags unless you explicitly specify tag filters. Additionally, if a job requires any other jobs (directly or indirectly), you must specify tag filters for those jobs.